src/pages/playlist/Playlist.jsx

```diff
@@ -1,44 +1,64 @@
 import "./playlist.css";
 import { Link, useNavigate } from "react-router-dom";
+import { getAllPlaylists, useVideoAction } from "../../contexts/index";
+import { useState } from "react";
 
 export function Playlist() {
-  const [playlist, setPlaylist] = useState([]);
-  async function getVideosList() {
-    try {
-      const { data } = await axios.get("/api/videos");
-      setPlaylist(() => data.videos);
-    } catch (e) {
-      console.log("error", e);
-    }
-  }
-  const playlistId = "";
+  const { playlist, dispatchAction } = useVideoAction();
+  const navigate = useNavigate();
 
-  useEffect(() => getVideosList(), []);
+  const [showModal, setShowModal] = useState(false);
+  useEffect(() => getAllPlaylists(dispatchAction, navigate), []);
   return (
     <>
       <Header />
       <SideNav />
       <div className="videolist-container d-grid">
-        <h4 className="text-center">Playlists</h4>
-        <ul className="custom-video-list list-no-bullet grid-gap playlist">
-          {playlist &&
-            playlist?.map(({ _id, title, videoId, author, thumnailHigh }) => {
-              return (
-                <Link to={`/explore/video/${_id}`} key={_id}>
-                  <div className="d-flex grid-gap playlist-card">
-                    <div>{"title title"}</div>
-                    <button className="btn btn-icon icon-trash">
-                      <MdDelete size={20} />
-                    </button>
-                  </div>
-                </Link>
-              );
-            })}
-        </ul>
+        <div className="d-flex playlist-header">
+          <h4 className="text-center">Playlists</h4>
+          <button
+            className="btn btn-primary"
+            onClick={() => {
+              setShowModal(true);
+            }}
+          >
+            CREATE NEW
+          </button>
+        </div>
+
+        {playlist?.length === 0 ? (
+          <div className="not-found">
+            <h5 className="text-center">Your have 0 playlists!</h5>
+            <div className="d-flex children-center img-not-found">
+              <img
+                src="https://cdn.iconscout.com/icon/free/png-128/playlist-3265179-2777597.png"
+                alt="playlist-img"
+              />
+            </div>
+          </div>
+        ) : (
+          <>
+            {" "}
+            <ul className="custom-video-list d-flex list-no-bullet playlist">
+              {playlist &&
+                playlist?.map(({ _id, title }) => {
+                  return (
+                    <Link to={`/playlists/${_id}`} key={_id}>
+                      <li className="d-flex grid-gap children-center playlist-card">
+                        {title}
+                      </li>
+                    </Link>
+                  );
+                })}
+            </ul>
+          </>
+        )}
+
+        {showModal && (
+          <PlaylistModal isAddToPlaylist={false} setShowModal={setShowModal} />
+        )}
```

Comment:
Make show modal a prop of Playlist Modal and conditionally render it. There by you are contorling the modal from outside but the content display logic is within the Modal Component

Reply:
i will do this later, for now I just went with the flow